applications/FluidDynamicsApplication/custom_elements/alternative_d_vms_dem_coupled.cpp

```diff
@@ -0,0 +1,999 @@
+    const unsigned int number_of_integration_points = gauss_weights.size();
+
+    TElementData data;
+    data.Initialize(*this,rCurrentProcessInfo);
+    for (unsigned int g = 0; g < number_of_integration_points; g++) {
+        this->UpdateIntegrationPointData(data, g, gauss_weights[g],row(shape_functions,g),shape_function_derivatives[g]);
+
+        this->UpdateSubscaleVelocityPrediction(data);
+    }
+}
+
+template <class TElementData>
+void AlternativeDVMSDEMCoupled<TElementData>::FinalizeNonLinearIteration(const ProcessInfo& rCurrentProcessInfo)
+{
+    // Get Shape function data
+    Vector gauss_weights;
+    Matrix shape_functions;
+    ShapeFunctionDerivativesArrayType shape_function_derivatives;
+    this->CalculateGeometryData(gauss_weights,shape_functions,shape_function_derivatives);
+    const unsigned int number_of_integration_points = gauss_weights.size();
+
+    TElementData data;
+    data.Initialize(*this,rCurrentProcessInfo);
+    for (unsigned int g = 0; g < number_of_integration_points; g++) {
+        this->UpdateIntegrationPointData(data, g, gauss_weights[g],row(shape_functions,g),shape_function_derivatives[g]);
+
+        this->UpdateSubscaleVelocity(data);
+    }
+}
+
+template< class TElementData >
+std::string AlternativeDVMSDEMCoupled<TElementData>::Info() const
+{
+    std::stringstream buffer;
+    buffer << "AlternativeDVMSDEMCoupled #" << this->Id();
+    return buffer.str();
+}
+
+
+template< class TElementData >
+void AlternativeDVMSDEMCoupled<TElementData>::PrintInfo(std::ostream& rOStream) const
+{
+    rOStream << "AlternativeDVMSDEMCoupled" << Dim << "D";
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Protected functions
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Evaluation of system terms on Gauss Points
+
+template< class TElementData >
+void AlternativeDVMSDEMCoupled<TElementData>::AlgebraicMomentumResidual(
+    const TElementData& rData,
+    const array_1d<double,3> &rConvectionVelocity,
+    array_1d<double,3>& rResidual) const
+{
+    const GeometryType rGeom = this->GetGeometry();
+
+    Vector convection; // u * grad(N)
+    this->ConvectionOperator(convection,rConvectionVelocity,rData.DN_DX);
+
+    const double density = this->GetAtCoordinate(rData.Density,rData.N);
+    const double viscosity = this->GetAtCoordinate(rData.DynamicViscosity, rData.N);
+    const auto& fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+    BoundedMatrix<double,Dim,Dim> permeability = this->GetAtCoordinate(rData.Permeability, rData.N);
+    BoundedMatrix<double,Dim,Dim> sigma = ZeroMatrix(Dim, Dim);
+
+    const auto& r_body_forces = rData.BodyForce;
+    const auto& r_velocities = rData.Velocity;
+    const auto& r_pressures = rData.Pressure;
+    const auto& fluid_fraction_gradient = this->GetAtCoordinate(rData.FluidFractionGradient, rData.N);
```

Comment:
```suggestion
    const auto& r_fluid_fraction_gradient = this->GetAtCoordinate(rData.FluidFractionGradient, rData.N);
```
(minor style guide suggestion)